Class Preferences

khaidoan2729 opened the pull request:
I resolved the conflicts caused after Theo's backend update (goated btw) and added GET endpoints. Now we can see the class preferences cards. 

<img width="1050" alt="Screen Shot 2025-01-30 at 13 53 25" src="https://github.com/user-attachments/assets/42246127-7bfa-4007-8c97-324048329d58" />

jjessieshang commented:
Hey Khai, im running into a 400 error for the following endpoint: GET /volunteer/class_preferences/d98889b0-2f2e-4172-b753-14cdccdd359c 400 121

jjessieshang commented on frontend/src/components/ClassPreferencesCard/index.js:
```diff
@@ -0,0 +1,65 @@
+function ClassPreferencesCard({ classData }) {
+
+    const RANK1_COLOR = "rgba(67, 133, 172, 1)";
+    const RANK2_COLOR = "rgba(67, 133, 172, 0.7)";
```
we can move these to the global styles file

jjessieshang commented on frontend/src/components/volunteerLayout/index.js:
```diff
@@ -115,6 +115,18 @@ function VolunteerLayout() {
+
+
+          <NavLink
+            to="/volunteer/class-preferences"
```
I'm assuming this is a temporary link to the page, however before we merge into main, let's put the temporary navigation somewhere on the volunteer profile page to match the design more closely and keep the navigation panel clean!
<img width="557" alt="Screenshot 2025-02-01 at 2 45 44 PM" src="https://github.com/user-attachments/assets/a2e9436a-bc39-4f75-8c60-b7eb3f2efd40" />

jjessieshang reviewed (commented):
great progress :)

khaidoan2729 commented:
- Added Class Preference Card on My Profile Page to navigate to Class Preferences Page. 
- Implemented GET function to retrieve preferences based on current volunteer ID and Class Preferences Card for displaying. 
- Implemented PUT function to update new preferences. 

New Components that might be useful in the future

### Modal

<img width="1022" alt="image" src="https://github.com/user-attachments/assets/8a34f609-f2dc-4c68-81f8-69cf2bbb217b" />

Props: 

- isOpen <bool>: to open (display) or close (hide) the modal.
- onClose <hook>: to trigger the close modal function.
- children
-  width <string>: modal width
- height <string>: modal height 

Example: 
```
const [modalOpen, setModalOpen] = useState(false);
const handleConfirm = () => {
  console.log("Confirming...");
}

return (
  <Modal isOpen={modalOpen} onClose={handleConfirm} width={"600px"} height={"90%"}>
    [children]
  </Modal>
);
```

### Checkbox
<img width="100" alt="image" src="https://github.com/user-attachments/assets/a1e0f5f1-19bf-412a-af0c-e8c45bee570e" />

Props: 

- onClicked <hook>: to trigger function when checkbox is clicked
- active <bool>: to determine whether the checkbox has been clicked. In the picture above, the upper checkbox is when active is false. 

Example: 
```
<Checkbox onClicked={()=>{handleCheckboxClicked(class_)}} active={true)}/>
```

jjessieshang reviewed (approved):
This looks so great Khai! Really great work on the frontend and backend :)

Let's rebase off main, squash commits, and merge this in, it will be great to have for the internal launch. 

Some points I have (that can be worked on in a future ticket):
- Discuss with designers how we can show which day of the week the schedule occurs on, as right now it only shows time
- Consider renaming the class_preferences table to be schedule_preferences as you are matching with schedule
